File: server/model/movie.js
const mongoose = require('mongoose');
const ratingSchema = require('./rating');
const commentSchema = require('./comments');
const movieSchema = new mongoose.Schema(
    {
        name:{
            type:String,
            required:true,
            trim:true
        },
        image:[{
          type:String,
          required:true
        }],
        category:[{
            type:String,
            required: true,
        }],
        characters:[{
            type:String,
            required:true
        }],
        description:{
            type:String,
            required:true
        },
        rating:[ratingSchema],
        comments:[commentSchema]
    }
);

const Movie = mongoose.model("Movie",movieSchema);
module.exports = {Movie , movieSchema};

File: lib/model/Movie.dart
import 'dart:convert';

import 'Rating.dart';

class Movie {
  final String name;
  final List<dynamic> images;
  final List<dynamic> categories;
  final List<dynamic> characters;
  final String description;
  final List<dynamic>? ratings;
  final String? id;
  List<dynamic>? comments;

  Movie(
      {required this.name,
      required this.images,
      required this.categories,
      required this.characters,
      required this.description,
      this.id,
      this.ratings,
      this.comments
      });


  factory Movie.fromJson(Map<String, dynamic> json) {

    return Movie(
      name: json["name"]!,
      images: List.of(json["image"]![0].replaceAll("[", "").replaceAll("]", "").split(",")).map((i) => i).toList(),
      categories:
          List.of(json["category"]![0].replaceAll("[", "").replaceAll("]", "").split(",")).map((i) => i).toList(),
      characters:
          List.of(json["characters"]![0].replaceAll("[", "").replaceAll("]", "").split(",")).map((i) => i).toList(),
      description: json["description"]!,
      ratings: List.of(json["rating"])
          .map((i) => i /* can't generate it properly yet */)
          .toList(),
      id: json["_id"],
      comments: List.of(json["comments"]).map((i) => i /* can't generate it properly yet */).toList(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      "name": name,
      "images": jsonEncode(images),
      "categories": jsonEncode(categories),
      "characters": jsonEncode(characters),
      "description": description,
      "ratings": jsonEncode(ratings),
      "id": id,
      "comments":jsonEncode(comments)
    };
  }
//


}
